nanoFramework.CoreInk/M5CoreInk.cs

```diff
@@ -0,0 +1,256 @@
+        private const int ScreenDCPin = 15;
+
+        #region properties
+
+        /// <summary>
+        /// Gets the upper button.
+        /// </summary>
+        public static GpioButton RollerLeft
+        {
+            get
+            {
+                _left ??= new(37, GpioController, false, PinMode.Input);
+
+                return _left;
+            }
+        }
+
+        /// <summary>
+        /// Gets the upper button.
+        /// </summary>
+        public static GpioButton RollerRight
+        {
+            get
+            {
+                _right ??= new(39, GpioController, false, PinMode.Input);
+
+                return _right;
+            }
+        }
+
+        /// <summary>
+        /// Gets the upper button.
+        /// </summary>
+        public static GpioButton RollerButton
+        {
+            get
+            {
+                _center ??= new(38, GpioController, false, PinMode.Input);
+
+                return _center;
+            }
+        }
+
+        /// <summary>
+        /// Gets the upper button.
+        /// </summary>
+        public static GpioButton Button
+        {
+            get
+            {
+                _button ??= new(5, GpioController, false);
+
+                return _button;
+            }
+        }
+
+        /// <summary>
+        /// Gets the power button.
+        /// </summary>
+        public static GpioButton Power
+        {
+            get
+            {
+                _power ??= new(27, GpioController, false);
+
+                return _power;
+            }
+        }
+
+        /// <summary>
+        /// Gets the green led.
+        /// </summary>
+        public static GpioPin Led
+        {
+            get
+            {
+                _led ??= GpioController.OpenPin(10, PinMode.Output);
+
+                return _led;
+            }
+        }
+
+        /// <summary>
+        /// Gets the Buzzer.
+        /// </summary>
+        public static Buzzer Buzzer
+        {
+            get
+            {
+                // SetPinFunction already made in the static constructor
+                _buzzer ??= new(2);
+
+                return _buzzer;
+            }
+        }
+
+        /// <summary>
+        /// Gets the main <see cref="GpioController"/>.
+        /// </summary>
+        public static GpioController GpioController
+        {
+            get
+            {
+                _gpio ??= new();
+
+                return _gpio;
+            }
+        }
+
+        /// <summary>
+        /// Gets the internal clock.
+        /// </summary>
+        public static Pcf8563 RTC
+        {
+            get
+            {
+                _rtc ??= new(_device);
+
+                return _rtc;
+            }
+        }
+
+        /// <summary>
+        /// Get the ePaper screen.
+        /// </summary>
+        public static Gdew0154m09 Screen
+        {
+            get
+            {
+                InitializeScreen();
```

Comment:
is there any way to skip initializing the screen if it's been already initialized?

Reply:
ok, forgot about it, just checked the function and it doesn't initialized it if not needed.